terraform/modules/cloudwatch/main.tf

```diff
@@ -0,0 +1,49 @@
+resource "aws_cloudwatch_event_rule" "ecs_scheduled_event" {
+  name        = local.cloudwatch_event_rule_name
+  description = "Scheduled event rule to run ECS task"
+  schedule_expression = var.cron_expression
+}
+
+resource "aws_cloudwatch_event_target" "ecs_event_target" {
+  rule      = aws_cloudwatch_event_rule.ecs_scheduled_event.name
+  target_id = "run-ecs-task"
+
+  arn = data.aws_ecs_cluster.existing_cluster.arn
+
+  ecs_target {
+    task_count          = 1
+    launch_type         = var.launch_type
+    platform_version    = "LATEST"
+    task_definition_arn = data.aws_ecs_task_definition.existing_task.arn
+  }
+}
+
+resource "aws_sns_topic" "slack_notifications" {
```

Comment:
SNS should be split out into a separate module. the topic can be created there and passed to the events module